bentoml/deployment/serverless/aws_lambda.py

```diff
@@ -125,119 +130,156 @@ def generate_handler_py(bento_name, apis, output_path):
     with open(os.path.join(output_path, "handler.py"), "w") as f:
         f.write(AWS_HANDLER_PY_TEMPLATE_HEADER.format(class_name=bento_name))
         for api in apis:
-            api_content = AWS_FUNCTION_TEMPLATE.format(api_name=api['name'])
+            api_content = AWS_FUNCTION_TEMPLATE.format(api_name=api.name)
             f.write(api_content)
     return
 
 
 class AwsLambdaDeploymentOperator(DeploymentOperatorBase):
     def apply(self, deployment_pb, repo, prev_deployment=None):
-        ensure_docker_available_or_raise()
-        deployment_spec = deployment_pb.spec
-        aws_config = deployment_spec.aws_lambda_operator_config
-
-        bento_path = repo.get(deployment_spec.bento_name, deployment_spec.bento_version)
-        bento_config = load_bentoml_config(bento_path)
-
-        template = 'aws-python3'
-        minimum_python_version = version.parse('3.0.0')
-        bento_python_version = version.parse(bento_config['env']['python_version'])
-        if bento_python_version < minimum_python_version:
-            template = 'aws-python'
-
-        with TemporaryServerlessContent(
-            archive_path=bento_path,
-            deployment_name=deployment_pb.name,
-            bento_name=deployment_spec.bento_name,
-            template_type=template,
-        ) as output_path:
-            generate_handler_py(
-                deployment_spec.bento_name, bento_config['apis'], output_path
-            )
-            generate_serverless_configuration_for_aws_lambda(
-                service_name=deployment_pb.name,
-                apis=bento_config['apis'],
-                output_path=output_path,
-                region=aws_config.region,
-                stage=deployment_pb.namespace,
+        try:
+            ensure_docker_available_or_raise()
+            deployment_spec = deployment_pb.spec
+            aws_config = deployment_spec.aws_lambda_operator_config
+
+            bento_path = repo.get(
+                deployment_spec.bento_name, deployment_spec.bento_version
             )
-            logger.info(
-                'Installing additional packages: serverless-python-requirements, '
-                'serverless-apigw-binary'
+            bento_config = load_bento_service_metadata(bento_path)
+
+            template = 'aws-python3'
+            minimum_python_version = version.parse('3.0.0')
+            bento_python_version = version.parse(bento_config.env.python_version)
+            if bento_python_version < minimum_python_version:
+                template = 'aws-python'
+
+            apis = generate_api_list(
```

Comment:
api_names = [aws_config.api_name] if aws_config.api_name is not None else bento_config.apis
verify_api_names(api_name, config)